WooCommerce/WooCommerceTests/ViewRelated/Orders/Order Creation/ProductSelectorViewModelTests.swift

```diff
@@ -1404,6 +1404,70 @@ final class ProductSelectorViewModelTests: XCTestCase {
         assertEqual(bundleProduct, productToConfigure)
     }
 
+    /// We call onMultipleSelectionCompleted to save the selections made on a modal product
+    /// selector, e.g. when an iPad is in split-screen mode and is rotated to landscape.
+    /// It's a blocking operation, so we only want to do it when required.
+    func test_updateSyncApproach_calls_onMultipleSelectionCompleted_when_approach_changes_to_immediate() {
+        // Given
+        let simpleProduct = Product.fake().copy(siteID: sampleSiteID, productID: 1, purchasable: true)
+        insert(simpleProduct)
+
+        let selectedItems = waitFor { promise in
+            let viewModel = ProductSelectorViewModel(siteID: self.sampleSiteID,
+                                                     selectedItemIDs: [],
+                                                     storageManager: self.storageManager,
+                                                     syncApproach: .onButtonTap,
+                                                     onMultipleSelectionCompleted: { selectedItems in
+                promise(selectedItems)
+            })
+
+            // When
+            viewModel.addSelection(id: 1)
+            viewModel.updateSyncApproach(to: .immediate)
+        }
+
+        // Then
+        assertEqual([1], selectedItems)
+    }
+
+    func test_updateSyncApproach_doesnt_call_onMultipleSelectionCompleted_when_approach_stays_on_immediate() {
+        // Given
+        let expectation = XCTestExpectation(description: "onMultipleSelectionCompleted called")
+        expectation.isInverted = true
```

Comment:
TIL `isInverted` , nice!

Reply:
Yeah me too 😅 I just figured there must be something.

Best used sparingly, as it needs to wait for the full length of `timeout` to know that it hasn't happened. I've reduced that to 0.1s, which works in my testing, when I force it to fail. The tests take about 25% of that time to fail on my machine, including everything before the wait, so it should be long enough to allow CI to fail when appropriate as well. cd8a200